internal/client/client_collapse.go

```diff
@@ -65,6 +66,15 @@ func (r reqCollapse) Close() error {
 }
 
 func (r reqCollapse) SendRequest(ctx context.Context, addr string, req *tikvrpc.Request, timeout time.Duration) (*tikvrpc.Response, error) {
+	if it := interceptor.GetRPCInterceptorFromCtx(ctx); it != nil {
```

Comment:
This means, when a request is collapsed and not called, the interceptor is still called. I'm not sure if it's the behavior you want...

Reply:
For our existing scene, this has no effect. But from your point of view, as a general Interceptor mechanism, is this behavior acceptable?
If you think this behavior is reasonable from a general design point of view, then it’s okay.

Reply:
I don't have a strong opinion on both... It depends on the use case...

Reply:
I think it's ok, and added a comment to explain this situation:

```go
//
// NOTE: Interceptor calls may not correspond one-to-one with the underlying gRPC requests.
// This is because there may be some exceptions, such as: request collapsed, request batched,
// no valid connection etc. If you have questions about the execution location of RPCInterceptor,
// please refer to: internal/client/client_collapse.go#SendRequest.
type RPCInterceptor func(next RPCInterceptorFunc) RPCInterceptorFunc
```